tools/release-plugin.sh

```diff
@@ -12,128 +13,437 @@ BASE=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+
+	yellow "Checking out prerelease branch."
+	# Is there an upstream prerelease branch already?
+	R=$( git ls-remote --heads origin prerelease )
+	if [[ -n "$R" ]]; then
+		error "There's an existing prerelease branch on GitHub!"
+		R=${R%%[ $'\t']*}
+		TMP=$( gh pr list --head=prerelease --state=open --json number,author,url,createdAt,title --jq '.[] | "\( .url ) - \( .createdAt | fromdateiso8601 | strftime( "%F %H:%IZ" ) ) - \( .title ) [\( .author.name )]"' )
+		if [[ -n "$TMP" ]]; then
+			echo "Someone probably needs to merge the following PR:"
+			echo "$TMP"
+			exit 1
+		fi
+		git fetch -q origin trunk "$R" || die "Something unexpected went wrong fetching the commit from GitHub"
+		if git diff --quiet origin/trunk..."$R"; then
+			echo "There don't seem to be any changes between that and trunk. If its author has abandoned the release it should be safe to delete the branch."
+			exit 1
+		fi
+		if ! git diff --quiet origin/trunk..."$R" '*/CHANGELOG.md'; then
+			echo "Looks like someone is in the middle of a release! Wait for them to merge the changes back into trunk."
+			exit 1
+		fi
+		echo "Looks like someone may be in the middle of a release! Wait for them to merge the changes back into trunk, or to abandon the release."
+		exit 1
+	fi
+
+	# Check out and push pre-release branch
+	if git rev-parse --verify prerelease &>/dev/null; then
+		proceed_p "Existing local prerelease branch found." "Delete it?" Y
+		git branch -D prerelease
+	fi
+
+	git checkout -b prerelease
+	if ! git push -u origin HEAD; then
+		send_tracks_event "jetpack_release_prerelease_push" '{"result": "failure"}'
+		die "Branch push failed. Check #jetpack-releases and make sure no one is doing a release already, then delete the branch at https://github.com/Automattic/jetpack/branches"
+	fi
+	GITBASE=$( git rev-parse --verify HEAD )
+	send_tracks_event "jetpack_release_prerelease_push" '{"result": "success"}'
+}
+
+function do_changelogs {
+	verify_prerelease_branch
+	# Loop through the projects and update the changelogs after building the arguments.
+	for PLUGIN in "${!PROJECTS[@]}"; do
+		yellow "Updating the changelog files for $PLUGIN."
+
+		# Add the PR numbers to the changelog.
+		ARGS=('-p')
+
+		# Add alpha and beta flags.
+		VERSION="${PROJECTS[$PLUGIN]}"
+		case $VERSION in
+			*-a* ) ARGS+=('-a');;
+			*-beta ) ARGS+=('-b');;
+		esac
+
+		# Explicitly pass the version number we want so there are no surprises.
+		ARGS+=( '-r' "${PROJECTS[$PLUGIN]}" )
+		ARGS+=("$PLUGIN");
+		tools/changelogger-release.sh "${ARGS[@]}"
+	done
+
+	# When it completes, wait for user to edit anything they want, then push key to continue.
+	read -r -s -p $'Edit and save all the changelog entries you want in a separate terminal or in the text editor of choice.\nCheck for consistency between the different entries, and keep in mind that your plugin changelog will be used in the plugin readme file.\n\nOnce you are happy with your work, press enter to continue the release process.'
```

Comment:
> or in the text editor of choice.

Grammar is off there. Possible fixes

```suggestion
	read -r -s -p $'Edit and save all the changelog entries you want in a separate terminal or in your text editor of choice.\nCheck for consistency between the different entries, and keep in mind that your plugin changelog will be used in the plugin readme file.\n\nOnce you are happy with your work, press enter to continue the release process.'
```
```suggestion
	read -r -s -p $'Edit and save all the changelog entries you want in a separate terminal or in the text editor of your choice.\nCheck for consistency between the different entries, and keep in mind that your plugin changelog will be used in the plugin readme file.\n\nOnce you are happy with your work, press enter to continue the release process.'
```